azimuth/dataset_split_manager.py

```diff
@@ -473,37 +490,38 @@ def _get_prediction_table(self, table_key: PredictionTableKey) -> Dataset:
             A table for this key, will create one if it doesn't exists.
         """
 
-        path = self._prediction_path(table_key=table_key)
-        if table_key not in self._prediction_tables and os.path.exists(path):
-            with FileLock(self._file_lock):
-                self._prediction_tables[table_key] = self.load_cache(path)
-        elif table_key not in self._prediction_tables:
+        pred_path = self._prediction_path(table_key=table_key)
+        if not os.path.exists(pred_path):
             empty_ds = Dataset.from_dict({"pred_row_idx": list(range(self.num_rows))})
             self._prediction_tables[table_key] = self._init_dataset_split(
                 empty_ds, self._prediction_tags
             ).remove_columns([DatasetColumn.row_idx])
             self.save_prediction_table(table_key)
+        else:
+            current_last_update = self._prediction_tables_last_update[table_key]
+            newest_pred_ds, last_update = self.load_latest_cache(pred_path, current_last_update)
+            if newest_pred_ds:
+                self._prediction_tables[table_key] = newest_pred_ds
+                self._prediction_tables_last_update[table_key] = last_update
```

Comment:
Detail, but we can avoid the `not` so there is less negation to understand.
```suggestion
        if os.path.exists(pred_path):
            current_last_update = self._prediction_tables_last_update[table_key]
            newest_pred_ds, last_update = self.load_latest_cache(pred_path, current_last_update)
            if newest_pred_ds:
                self._prediction_tables[table_key] = newest_pred_ds
                self._prediction_tables_last_update[table_key] = last_update
        else:
            empty_ds = Dataset.from_dict({"pred_row_idx": list(range(self.num_rows))})
            self._prediction_tables[table_key] = self._init_dataset_split(
                empty_ds, self._prediction_tags
            ).remove_columns([DatasetColumn.row_idx])
            self.save_prediction_table(table_key)
```

Reply:
Good point!